src/main/java/org/synyx/urlaubsverwaltung/absence/AbsencePeriod.java

```diff
@@ -0,0 +1,272 @@
+ *     <li>31.March 2021 (vacation morning)</li>
+ *     <li>9.June 2021 (vacation noon)</li>
+ *     <li>26.August 2021 to 27.August 2021 (sick full day)</li>
+ * </ul>
+ */
+public class AbsencePeriod {
+
+    public enum AbsenceType {
+        VACATION,
+        SICK,
+    }
+
+    public enum AbsenceStatus {
+        // vacation
+        WAITING,
+        TEMPORARY_ALLOWED,
+        ALLOWED,
+        ALLOWED_CANCELLATION_REQUESTED,
+        // sick note
+        ACTIVE,
+    }
+
+    private final Person person;
+    private final List<AbsencePeriod.Record> absenceRecords;
+
+    public AbsencePeriod(Person person, List<Record> absenceRecords) {
+        this.person = person;
+        this.absenceRecords = absenceRecords;
+    }
+
+    public Person getPerson() {
+        return person;
+    }
+
+    public List<AbsencePeriod.Record> getAbsenceRecords() {
+        return Collections.unmodifiableList(absenceRecords);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        AbsencePeriod that = (AbsencePeriod) o;
+        return Objects.equals(person, that.person) && Objects.equals(absenceRecords, that.absenceRecords);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(person, absenceRecords);
+    }
+
+    @Override
+    public String toString() {
+        return "AbsencePeriod{" +
+            "person=" + person +
+            ", absenceRecords=" + absenceRecords +
+            '}';
+    }
+
+    /**
+     * Specifies an absence for one date. The absence consists of `morning` and `evening`.
+     * You may have to handle information yourself for "full absence vacation". In This case morning and evening are
+     * defined.
+     */
+    public static class Record {
+
+        private final LocalDate date;
+        private final Person person;
+        private final RecordMorning morning;
+        private final RecordNoon noon;
+
+        public Record(LocalDate date, Person person, RecordMorning morning) {
+            this.date = date;
+            this.person = person;
+            this.morning = morning;
+            this.noon = null;
+        }
+
+        public Record(LocalDate date, Person person, RecordNoon noon) {
+            this.date = date;
+            this.person = person;
+            this.morning = null;
+            this.noon = noon;
+        }
+
+        public Record(LocalDate date, Person person, RecordMorning morning, RecordNoon noon) {
+            this.date = date;
+            this.person = person;
+            this.morning = morning;
+            this.noon = noon;
+        }
+
+        public LocalDate getDate() {
+            return date;
+        }
+
+        public Person getPerson() {
+            return person;
+        }
+
+        public boolean isHalfDayAbsence() {
+            return this.morning == null || this.noon == null;
```

Comment:
return (this.morning == null && this.noon != null) || (this.morning != null && this.noon == null);
??

Reply:
Hab noch tests geschrieben und es so formuliert wie oben.